plugins/azure/appservice/appServiceDiagnosticLogs.js

```diff
@@ -0,0 +1,61 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'App Service Diagnostic Logging Enabled',
+    category: 'App Service',
+    domain: 'Application Integration',
+    description: 'Ensures that diagnostic logging is enabled for Azure App Service.',
+    more_info: 'Enabling Azure App Service Diagnostics Logging provides a quick and easy way to view application logs, allowing users to diagnose and resolve issues, including errors, performance bottlenecks, and security concerns.',
+    recommended_action: 'Enable diagnostic logging for all App Services.',
+    link: 'https://learn.microsoft.com/en-us/azure/app-service/troubleshoot-diagnostic-logs',
+    apis: ['webApps:list', 'diagnosticSettings:listByAppServices'],
+    realtime_triggers: ['microsoftweb:sites:write', 'microsoftinsights:diagnosticsettings:write', 'microsoftinsights:diagnosticsettings:delete', 'microsoftweb:sites:delete'],
+   
+    run: function(cache, settings, callback) {
+        const results = [];
+        const source = {};
+        const locations = helpers.locations(settings.govcloud);
+        
+        async.each(locations.webApps, (location, rcb) => {
+
+            const webApps = helpers.addSource(cache, source,
+                ['webApps', 'list', location]);
+
+            if (!webApps) return rcb();
+
+            if (webApps.err || !webApps.data) {
+                helpers.addResult(results, 3, 'Unable to query for Web Apps: ' + helpers.addError(webApps), location);
+                return rcb();
+            }
+
+            if (!webApps.data.length) {
+                helpers.addResult(results, 0, 'No existing Web Apps found', location);
+                return rcb();
+            }
+
+            webApps.data.forEach(webApp => {
+                if (!webApp.id) return;
+                const diagnosticSettings = helpers.addSource(cache, source,
+                    ['diagnosticSettings', 'listByAppServices', location, webApp.id]);
+                if (!diagnosticSettings || diagnosticSettings.err || !diagnosticSettings.data) {
```

Comment:
```suggestion

                if (!diagnosticSettings || diagnosticSettings.err || !diagnosticSettings.data) {
```
add line break before/after if statements